broker/src/main/java/io/zeebe/broker/system/configuration/NetworkCfg.java

```diff
@@ -74,11 +74,18 @@ public void setPortOffset(final int portOffset) {
     this.portOffset = portOffset;
   }
 
-  public ByteValue getMaxMessageSize() {
-    return new ByteValue(maxMessageSize);
+  public String getMaxMessageSize() {
+    return maxMessageSize;
+  }
+
+  public long getMaxMessageSizeInBytes() {
+    return ByteValueParser.fromString(maxMessageSize).toBytes();
   }
 
   public void setMaxMessageSize(final String maxMessageSize) {
+    // call parsing logic to provoke any exceptions that might occur during parsing
+    ByteValueParser.fromString(maxMessageSize);
```

Comment:
Nit: not sure if necessary as when we parse the config file, we could have an invalid string and we'll anyway only figure it out when we try to access it via `getMaxMessageSizeInBytes`.

Reply:
The idea here (and in other places) is to fail early and in a defined phase of computation. 

Setting the configuration with an invalid value will throw an exception during startup, so it would be noticed right away. Throwing an exception whenever we arbitrarily access a setting seems more risky to me.

There is also a school of thought which claims that getters should never throw exceptions.

I would strongly argue to keep it as it is. Also, if we use Spring Boot, we would change this interface to accept `DataSize` instead of `String`, and remove the parsing altogether. But the net behavior would be the same: exception will be thrown while parsing the configuration, not while accessing it.

Reply:
Could we store the returned value ? Then we do not need to re-calculate it again later? Do we need the string in the Cfg anyway?

Reply:
Yes we could store the returned value. I kinda opted against it, though. This is more a preparatory step to later replace `String` with `DataSize` (from Spring Boot). In that later stage I would store the `DataSize` value because it's the ground truth. The byte value is just a calculated value based on that. In that setup, the parsing will have happened before the `DataSize` object is created, so the expensive step is only done once. Also, it didn't look like the getter is called with a high frequency, so I think the performance gains of caching the result would be negligible.

Reply:
Ok so this PR is more a transitive state to the new cfg. Thanks for clarification